test/IsolatedKernelTestCase.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Matthias\BundlePlugins\Tests;
+
+/**
+ * Class IsolatedKernelTestCase
+ *
+ * @package Matthias\BundlePlugins\Tests
+ */
+class IsolatedKernelTestCase extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var string Directory with symfony cache
+     */
+    private $tempDirectory;
+
+    private $usedIdentifiers = array();
+
+    /**
+     * Sets a temporary directory for storing symfony cache
+     */
+    public function setUp()
+    {
+        parent::setUp();
+        $this->tempDirectory = tempnam(sys_get_temp_dir(), 'symfony-phpunit');
+
+        // We don't want a temporary _file_, just a nice and unique path
+        // If you can't stand this uglyness, please provide a pull request ;-)
+        unlink($this->tempDirectory);
+    }
+
+    /**
+     * @param array $configuration
+     * @param array $plugins
+     *
+     * @return AppKernel
+     */
+    protected function createKernel($configuration, $plugins)
+    {
+        $kernel = new AppKernel(
+            $configuration,
+            $plugins,
+            mt_rand(0, 9999999),
+            $this->getCacheDir(),
+            $this->getLogsDir()
+        );
+        $kernel->boot();
+        return $kernel;
+    }
+
+    /**
+     * @return string
+     */
+    private function getCacheDir()
+    {
+        return $this->tempDirectory . '/cache';
+    }
+
+    /**
+     * @return string
+     */
+    private function getLogsDir()
+    {
+        return $this->tempDirectory . '/logs';
+    }
+
+    public function tearDown()
+    {
+        // TODO: Are we responsible to remove again? Temp dirs get cleaned upon reboot anyway
```

Comment:
The process itself won't remove it, since it's considered a cache by Symfony. I was doubting to remove everything from the temp folder. I'm a bit 'scared' to implement recusive remove in this test, since several projects implementing something like that in the past, have shown that may be 'suboptimal'  when a bug occurs :)